cylc/flow/scheduler.py

```diff
@@ -1634,8 +1595,10 @@ async def update_data_structure(self):
             # Publish updates:
             if self.data_store_mgr.publish_pending:
                 self.data_store_mgr.publish_pending = False
-                await self.publisher.publish(
+                self.server.publish_queue.put(
                     self.data_store_mgr.publish_deltas)
+                # Yield control to other threads
+                sleep(0)
```

Comment:
As above
```suggestion
                await asyncio.sleep(0)
```

Reply:
Just as the comment above it suggests:

`time.sleep` is used to yield control to other threads (halting the current thread).. Whereas, I believe `asyncio.sleep` just allows other ayncio execution for the duration.

Reply:
The distinction of yielding to event loop vs other threads is a bit subtle imho, something I missed the first time round. I think it pays to be explanatory about it, something like
```python
# Non-async sleep - yield to other threads rather than event loop

Reply:
Well, I commented what we're doing with the `time.sleep`... 
What's `asyncio.sleep` got to do with it?
Why would devs assume they're the same and that it needs to be changed?

Reply:
> What's `asyncio.sleep` got to do with it?

Because most patterns involving a `while True` inside an async function should have `asyncio.sleep(short_duration)` in them to yield to the event loop.

As you've explained, in this case we want to yield to other threads instead. But the fact that it took some explanation for me to notice the difference is why I'm requesting a comment along those lines.

Reply:
Amended.. I appreciated your code hygiene 👍